File: Udacity - interview prepa course practise/NumLoopCounter.java
/*
Calculate the number of "loops" in a number.
A number has a loop when it has a closed circle when you draw it, so
6 has one loop, 1 has no loop and 8 has two loops.
So 2876 has three loops

*/

import java.util.HashMap;

class NumLoopCounter{
public static int loopCounter(int num){
    HashMap<Integer, Integer> numAndValue =
    new HashMap<>();

    numAndValue.put(0,0);
    numAndValue.put(1,0);
    numAndValue.put(2,0);
    numAndValue.put(3,0);
    numAndValue.put(4,0);
    numAndValue.put(5,0);
    numAndValue.put(6,1);
    numAndValue.put(7,0);
    numAndValue.put(8,2);
    numAndValue.put(9,1);

    String numsString = num + "";
    int counter = 0;
    for (int i = 0; i < numsString.length(); i++) {
        counter += numAndValue.get(Integer.parseInt(String.valueOf(numsString.charAt(i))));
    }

    return counter;
  }

  public static void main(String [] args){
    System.out.print(loopCounter(2876));
  }
}

File: Udacity/num_loop_counter.py
"""
Calculate the number of "loops" in a number.
A number has a loop when it has a closed circle when you draw it, so
6 has one loop, 1 has no loop and 8 has two loops.
So 2876 has three loops

"""

def loop_counter(num):
    num_and_value = {0:0, 1:0, 2:0, 3:0, 4:0, 5:0, 6:1, 7:0, 8:2, 9:1}

    counter = 0
    num_str = str(num)
    for i in range(len(num_str)):
        if num_str[i].isdigit():
            counter += num_and_value.get(int(num_str[i]))

    return counter

counter = loop_counter(2876)
print(counter)
